lib/cqm/converter/qdm_patient.rb

```diff
@@ -238,34 +241,34 @@ def unpack_diagnosis(hds_attrs)
       end
       # Remove diagnosis if principalDiagnosis is equivalent.
       return unless hds_attrs.key?('diagnosis') && hds_attrs.key?('principalDiagnosis')
-      return unless hds_attrs['diagnosis']['values'] && hds_attrs['diagnosis']['values'].first == hds_attrs['principalDiagnosis']
+      return unless hds_attrs['diagnosis']['values'] && Hash[hds_attrs['diagnosis']['values'].first.sort] == Hash[hds_attrs['principalDiagnosis'].sort]
```

Comment:
This one too. Was the second condition never returning `true` the way prior?

Reply:
The way prior was for underscore variable names. This change makes sure the hash keys are sorted before comparison.